crates/swift-bridge-ir/src/codegen/codegen_tests/option_codegen_tests.rs

```diff
@@ -460,6 +460,122 @@ void* __swift_bridge__$some_function(void);
     }
 }
 
+/// Test code generation for Rust function that returns an Option<&OpaqueRustType>
+mod extern_rust_fn_return_option_ref_opaque_rust_type {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod ffi {
+                extern "Rust" {
+                    type SomeType;
+                    fn some_function () -> Option<&SomeType>;
+                }
+            }
+        }
+    }
+
+    fn expected_rust_tokens() -> ExpectedRustTokens {
+        ExpectedRustTokens::Contains(quote! {
+            #[export_name = "__swift_bridge__$some_function"]
+            pub extern "C" fn __swift_bridge__some_function() -> *const super::SomeType {
+                if let Some(val) = super::some_function() {
+                    val as *const super::SomeType
+                } else {
+                    std::ptr::null()
+                }
+            }
+        })
+    }
+
+    fn expected_swift_code() -> ExpectedSwiftCode {
+        ExpectedSwiftCode::ContainsAfterTrim(
+            r#"
+func some_function() -> Optional<SomeTypeRef> {
+    { let val = __swift_bridge__$some_function(); if val != nil { return SomeTypeRef(ptr: val!) } else { return nil } }()
+}
+"#,
+        )
+    }
+
+    fn expected_c_header() -> ExpectedCHeader {
+        ExpectedCHeader::ContainsAfterTrim(
+            r#"
+void* __swift_bridge__$some_function(void);
+    "#,
+        )
+    }
+
+    #[test]
+    fn extern_rust_fn_return_option_opaque_rust_type() {
+        CodegenTest {
+            bridge_module: bridge_module_tokens().into(),
+            expected_rust_tokens: expected_rust_tokens(),
+            expected_swift_code: expected_swift_code(),
+            expected_c_header: expected_c_header(),
+        }
+        .test();
+    }
+}
+
+/// Test code generation for Rust function that returns an Option<&OpaqueRustType>
+mod extern_rust_fn_arg_option_ref_opaque_rust_type {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod ffi {
+                extern "Rust" {
+                    type SomeType;
+                    fn some_function (arg: Option<&SomeType>) -> Option<&SomeType>;
```

Comment:
I think this comment got swallowed since I was commenting on orphaned commits, so let me try here in the PR's "files changed" page.

Can remove the `-> Option<&SomeType>` here.

I made a mistake when I said to add it. The return case is already tested in the other test that you added so we don't need it here.

Reply:
So remove the entire module then? or...

Reply:
Keep the module but remove the `-> Option<&SomeType>` from this line 530 right here

So ` fn some_function (arg: Option<&SomeType>) -> Option<&SomeType>;`

becomes ` fn some_function (arg: Option<&SomeType>);